src/js/angular/ttyg/controllers/ttyg-view.controller.js

```diff
@@ -553,12 +553,14 @@ function TTYGViewCtrl(
      * @param {ChatModel} chat - the chat to be deleted.
      */
     const onDeleteChat = (chat) => {
+        TTYGContextService.deleteChat(chat);
```

Comment:
Why? Shouldn't you remove the chat from the context when its actually deleted, and not removing it before the request and the adding it back if operation fails??